app/v2/services/RetrieveHistoricNonFhlUkPropertyPeriodSummaryService.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.services
+
+import cats.data.EitherT
+import cats.implicits._
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.Logging
+import v2.connectors.RetrieveHistoricNonFhlUkPropertyPeriodSummaryConnector
+import v2.controllers.EndpointLogContext
+import v2.models.errors._
+import v2.models.request.retrieveHistoricNonFhlUkPiePeriodSummary.RetrieveHistoricNonFhlUkPiePeriodSummaryRequest
+import v2.models.response.retrieveHistoricNonFhlUkPiePeriodSummary.RetrieveHistoricNonFhlUkPiePeriodSummaryResponse
+import v2.support.DownstreamResponseMappingSupport
+
+import javax.inject.{ Inject, Singleton }
+import scala.concurrent.{ ExecutionContext, Future }
+
+@Singleton
+class RetrieveHistoricNonFhlUkPropertyPeriodSummaryService @Inject()(connector: RetrieveHistoricNonFhlUkPropertyPeriodSummaryConnector)
+    extends DownstreamResponseMappingSupport
+    with Logging {
+
+  private val downstreamErrorMap =
```

Comment:
To avoid duplication of these error codes in both code + tests, can we move these error code map to a companion object or a trait and then either import or extend the trait, as its duplicated in the test code.